src/main/java/hudson/plugins/jacoco/JacocoPublisher.java

```diff
@@ -240,14 +274,103 @@ public boolean isChangeBuildStatus() {
     public boolean getChangeBuildStatus() {
 		return changeBuildStatus;
 	}
-	
+    @DataBoundSetter
+    public void setExecPattern(String execPattern) {
+        this.execPattern = execPattern;
+    }
+
+    @DataBoundSetter
+    public void setClassPattern(String classPattern) {
+        this.classPattern = classPattern;
+    }
+
+    @DataBoundSetter
+    public void setSourcePattern(String sourcePattern) {
+        this.sourcePattern = sourcePattern;
+    }
+
+    @DataBoundSetter
+    public void setMinimumInstructionCoverage(String minimumInstructionCoverage) {
+        this.minimumInstructionCoverage = checkThresholdInput(minimumInstructionCoverage);
+    }
+
+    @DataBoundSetter
+    public void setMinimumBranchCoverage(String minimumBranchCoverage) {
+        this.minimumBranchCoverage = checkThresholdInput(minimumBranchCoverage);
+    }
+
+    @DataBoundSetter
+    public void setMinimumComplexityCoverage(String minimumComplexityCoverage) {
+        this.minimumComplexityCoverage = checkThresholdInput(minimumComplexityCoverage);
+    }
+
+    @DataBoundSetter
+    public void setMinimumLineCoverage(String minimumLineCoverage) {
+        this.minimumLineCoverage = checkThresholdInput(minimumLineCoverage);
+    }
+
+    @DataBoundSetter
+    public void setMinimumMethodCoverage(String minimumMethodCoverage) {
+        this.minimumMethodCoverage = checkThresholdInput(minimumMethodCoverage);
+    }
+
+    @DataBoundSetter
+    public void setMinimumClassCoverage(String minimumClassCoverage) {
+        this.minimumClassCoverage = checkThresholdInput(minimumClassCoverage);
+    }
+
+    @DataBoundSetter
+    public void setMaximumInstructionCoverage(String maximumInstructionCoverage) {
+        this.maximumInstructionCoverage = checkThresholdInput(maximumInstructionCoverage);
+    }
+
+    @DataBoundSetter
+    public void setMaximumBranchCoverage(String maximumBranchCoverage) {
+        this.maximumBranchCoverage = checkThresholdInput(maximumBranchCoverage);
+    }
+
+    @DataBoundSetter
+    public void setMaximumComplexityCoverage(String maximumComplexityCoverage) {
+        this.maximumComplexityCoverage = checkThresholdInput(maximumComplexityCoverage);
+    }
+
+    @DataBoundSetter
+    public void setMaximumLineCoverage(String maximumLineCoverage) {
+        this.maximumLineCoverage = checkThresholdInput(maximumLineCoverage);
+    }
+
+    @DataBoundSetter
+    public void setMaximumMethodCoverage(String maximumMethodCoverage) {
+        this.maximumMethodCoverage = checkThresholdInput(maximumMethodCoverage);
+    }
+
+    @DataBoundSetter
+    public void setMaximumClassCoverage(String maximumClassCoverage) {
+        this.maximumClassCoverage = checkThresholdInput(maximumClassCoverage);
+    }
+
+    @DataBoundSetter
+    public void setChangeBuildStatus(boolean changeBuildStatus) {
```

Comment:
Let's open a new PR to not mix things in this PR and maintain it as minimal as possible, just having minimal changes to make it work in pipeline